vpr/src/util/vpr_utils.cpp

```diff
@@ -2184,10 +2217,41 @@ int get_logical_pin(t_physical_tile_type_ptr physical_tile,
     return result->second.pin;
 }
 
+int get_physical_pin(int sub_tile_index,
+                     t_physical_tile_type_ptr physical_tile,
+                     t_logical_block_type_ptr logical_block,
+                     int pin) {
```

Comment:
What is the difference between these two versions of get_physical_pins()?

Reply:
Basically, one is the superset of the other.

I have modified as follows;
- The one with the sub_tile_index parameter is the one that performs the lookup to get the physical pin.
- The other homonym function, as first thing gets the sub tile index based on which sub tile is compatible with the logical block, then it calls the `get_physical_pin` with the sub_tile_index parameter.

This was done mainly because now, in the `logical pin <-> physical pin` mapping there is the sub tile concept introduced. The logical pin now has a sub tile index.

During clustering there is the need to access the physical pin related to a logical one, but it is not important which exact sub tile is needed, as the physical pin is needed only to check the type of pin for instance.

The only important place where the sub tile index needs to be the correct one, is during the block sync at the end of placement, where the "real" physical pins are mapped against the logical pins.